http.go

```diff
@@ -347,8 +337,39 @@ func Get(w http.ResponseWriter, req *http.Request, store mdata.Store, defCache *
 
 	js := bufPool.Get().([]byte)
 	if legacy {
-		sort.Sort(SeriesByTarget(out))
-		js, err = graphiteJSON(js, out)
+		//check for duplicate series names. If found merge the results.
+		seriesByTarget := make(map[string][]Series)
+		for _, series := range out {
+			if _, ok := seriesByTarget[series.Target]; !ok {
+				seriesByTarget[series.Target] = make([]Series, 0)
+			}
+			seriesByTarget[series.Target] = append(seriesByTarget[series.Target], series)
+		}
+		merged := make([]Series, len(seriesByTarget))
+		count := 0
+		for _, series := range seriesByTarget {
```

Comment:
just iterate using `for i, series` and use i instead of count. simpler and easier to understand what the variable is used for